packages/eslint/rules/no-throw-in-sentry-scope.js

```diff
@@ -0,0 +1,51 @@
+module.exports = {
+   meta: {
+      type: 'suggestion',
+      fixable: 'code',
+      docs: {
+         description: 'Ensure Sentry.withScope callback never calls throw',
+         category: 'Best Practices',
+         recommended: true,
+      },
+   },
+   create: function (context) {
+      let isSentryWithScope = false;
+
+      return {
+         CallExpression(node) {
+            if (
+               node.callee.type === 'MemberExpression' &&
+               node.callee.object &&
+               node.callee.object.name === 'Sentry' &&
+               node.callee.property &&
+               node.callee.property.name === 'withScope'
+            ) {
+               isSentryWithScope = true;
+            }
+         },
+         'CallExpression:exit': function (node) {
```

Comment:
I don't understand why we would need this, the expression either has scope or not no? Why do we need to check at the end of the expression?

Reply:
Ah, this is to allow checking nested functions.

An example:

```
const func1 = () => {
   throw new Error();
};

const func2 = () => {
   func1();
};


Sentry.withScope(scope => {
  func2();
});
```

I am looking for the withScope the marking it as such. We also have to manually unmark it in eslint once we leave the withScope function call.